### PR TITLE
Fixes being able to screw the taser into ED209s multiple times

### PR DESCRIPTION
**Pull request by Haevacht**
Replaces a sleep with a do_after.

Incorporates tool speed because I can.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/bot/construction.dm [hunk 1]
```diff
@@ -177,7 +177,7 @@
 				playsound(loc, W.usesound, 100, 1)
 				var/turf/T = get_turf(user)
 				user << "<span class='notice'>You start attaching the gun to the frame...</span>"
-				sleep(40)
+				do_after(user, 40*W.toolspeed, 0, src, 1)
```
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/bot/construction.dm]: It has to be if(do_after...)

And everything under it indented

Comment (Haevacht) [inline code/modules/mob/living/simple_animal/bot/construction.dm]: On it.

Comment (Haevacht) [inline code/modules/mob/living/simple_animal/bot/construction.dm]: Seemed to work as it was, I assume this makes 100% sure the do_after completes before progressing?

File: code/modules/mob/living/simple_animal/bot/construction.dm [hunk 2]
```diff
@@ -177,11 +177,11 @@
 				playsound(loc, W.usesound, 100, 1)
 				var/turf/T = get_turf(user)
 				user << "<span class='notice'>You start attaching the gun to the frame...</span>"
-				sleep(40)
-				if(get_turf(user) == T)
-					build_step++
-					name = "armed [name]"
-					user << "<span class='notice'>Taser gun attached.</span>"
+				if(do_after(user, 40*W.toolspeed, 0, src, 1))
+					if(get_turf(user) == T)
```
Comment (AnturK) [inline code/modules/mob/living/simple_animal/bot/construction.dm]: This check is redundant now, do after already checks user's movement.

**Pull request merged**